Add render_node() to render elements without parents.

pelme opened the pull request:
This is an attempt at making it possible to render elements without a parent element.

- I do like fragments as plain lists and am not keen on introducing a Fragment element
- I also like that you typically can just `print()` or `str()` a htpy element and it gets rendered without having to import a render function. I think that should be the standard way of triggering a render. This function is therefore intentionally named `render_orphans` and not `render` to not confuse anyone into using it when a plain `str()` call would do.

What do you think about this @zulqasar @davepeck ?

References:
- #22 
- https://github.com/pelme/htpy/discussions/23#discussioncomment-9610282

davepeck commented:
I like this direction.

(A few weeks ago, [in our earlier discussion](https://github.com/pelme/htpy/discussions/23#discussioncomment-9610282), I mentioned that I thought I had a preference for a `render(...)` method rather than introducing an explicit `Fragment` type. Apparently, I still do!)

I wonder if it makes sense to tie the method directly to the `Node` type? aka call it `render_node(node: Node) -> str`. That would sort of complete the `htpy` type story: either you have something that derives from `BaseElement`, in which case, call `str(my_thing)`; otherwise, call `render_node(...)`. And, uh, feel free to call `render_node(...)` even if you know you've got a `BaseElement`.

I guess this approach could leave some lingering questions about escaping behavior, maybe?

pelme commented:
Hi @davepeck , thanks for your feedback. I have been contemplating about this and I think your suggestion makes sense.

I have updated the PR to name it `render_node()` and also added `iter_node()` which does the equivalent for iteration. As you pointed out, I think this completes the type story. The implementation was trivial since `iter_node` already did exist internally as `_iter_children` which arguably is a worse name. :) 

The docs only mention `render_node()` in one place and there is a note to avoid it for regular elements. I hope that is enough to prevent people from using it instead of str().

pelme commented:
`render_node` / `iter_node` was committed via 0c07a5a99c40f88ba4266e9b9de7b179adc4f225 and released in `24.7.2`!
